src/digitaltwin/s3_connection.py

```diff
@@ -0,0 +1,228 @@
+# -*- coding: utf-8 -*-
+"""This script defines a python class to facilitate interactions with an Amazon S3 bucket."""
+
+import logging
+import pathlib
+import pickle
+from io import BytesIO
+from typing import List, Union
+
+import boto3
+import geopandas as gpd
+import networkx as nx
+import xarray as xr
+from pyproj import CRS
+
+from src.config import EnvVariable
+
+log = logging.getLogger(__name__)
+
+
+class S3Manager:
+    """
+    A class for managing interactions with an Amazon Simple Storage Service (Amazon S3) bucket.
+    Provides methods to interact with an AWS S3 bucket, including storing and retrieving objects,
+    listing objects, removing objects, uploading files, retrieving files, and clearing the entire bucket.
+    """
+
+    def __init__(self) -> None:
+        """
+        Initialize an S3Manager instance.
+        Sets up the S3Manager with the necessary AWS credentials obtained from environment variables and creates a boto3
+        session, S3 client, and S3 resource for interacting with an Amazon Simple Storage Service (Amazon S3) bucket.
+        """
+        self.access_key_id = EnvVariable.AWS_ACCESS_KEY_ID
+        self.secret_access_key = EnvVariable.AWS_SECRET_ACCESS_KEY
+        self.bucket_name = EnvVariable.AWS_BUCKET_NAME
+        self.session = self._create_session()
+        self.s3_client = self.session.client("s3")
+        self.s3_resource = self.session.resource("s3")
+
+    def _create_session(self) -> boto3.session.Session:
+        """
+        Creates a boto3 session using the provided AWS credentials.
+
+        Returns
+        -------
+        boto3.session.Session
+            A boto3 session that allows interaction with AWS services.
+        """
+        # Create a boto3 session for interacting with AWS services
+        session = boto3.Session(
+            aws_access_key_id=self.access_key_id,
+            aws_secret_access_key=self.secret_access_key)
+        return session
```

Comment:
Same comment as in newzealidar:

I am unsure if `boto3.Session` has good error messages but it might be worth doing some explicit validation that `access_key_id` and `secret_access_key` aren't empty. See what happens when they're empty and make a decision on if it's needed.